app/services/character/relocator.rb

```diff
@@ -0,0 +1,115 @@
+# Steps:
+# validate single user, unshared icons, galleries, and templates (if included), lack of character groups
+# collect: templates (if included), characters, posts, replies, drafts, galleries, icons
+# update all on unaudited (templates, aliases, drafts, galleries, icons)
+# UpdateModelJob on audited (characters, posts, replies)
+
+class Character::Relocator < Object
+  def initialize(admin_id)
+    @audited_user_id = admin_id
+  end
+
+  def transfer(character_ids, new_user_id, include_templates: false)
+    @characters = Character.where(id: character_ids)
+    @user_id = validate_user!
+    @user_other_characters = Character.where(user_id: @user_id).where.not(id: @characters.ids)
+    @other_character_galleries = get_galleries(@user_other_characters)
+    @other_character_icon_ids = get_icon_ids(@user_other_characters, @other_character_galleries)
+
+    raise CharacterGroupError.new('Characters must not have groups') if @characters.where.not(character_group: nil).exists?
+    @galleries = validate_galleries!
+    @icons = validate_icons!
+    @templates = validate_templates! if include_templates
+
+    @posts = Post.where(character_id: character_ids, user_id: @user_id)
+    @replies = Reply.where(character_id: character_ids, user_id: @user_id)
+    @drafts = ReplyDraft.where(character_id: character_ids, user_id: @user_id)
+
+    User.find(new_user_id)
```

Comment:
?

Reply:
Making sure the user actually exists, iirc

Reply:
Yeah but we do this in validate_user! as well

Reply:
I think that's the old user though?

Reply:
yeah validate_user! is checking the characters all have one user who exists, this is checking that the user_id the service was given corresponds to an actual user